Purchasing.Mvc/Controllers/WorkgroupController.cs

```diff
@@ -479,6 +480,121 @@ public async Task<ActionResult> AddAccount(int id, WorkgroupAccount workgroupAcc
             return View(viewModel);
         }
 
+        [HttpGet]
+        public ActionResult UpdateAccountsFromFinjector(int id)
+        {
+            var workgroup = _workgroupRepository.GetNullableById(id);
+            if (workgroup == null)
+            {
+                ErrorMessage = "Workgroup could not be found";
+                return this.RedirectToAction(nameof(WorkgroupController.Index), typeof(WorkgroupController).ControllerName(), new { showAll = false });
+            }
+
+            if (workgroup.Administrative)
+            {
+                ErrorMessage = "Accounts may not be added to an administrative workgroup.";
+                return this.RedirectToAction(nameof(Details), new { id = id });
+            }
+
+            return View(workgroup);
+        }
+
+        [HttpPost]
+        public ActionResult UpdateAccountsFromFinjector(int id, string chartsString, bool onlyAdd = false)
+        {
+            var workgroup = _workgroupRepository.GetNullableById(id);
+            if (workgroup == null)
+            {
+                ErrorMessage = "Workgroup could not be found";
+                return this.RedirectToAction(nameof(WorkgroupController.Index), typeof(WorkgroupController).ControllerName(), new { showAll = false });
+            }
+
+            if (workgroup.Administrative)
+            {
+                ErrorMessage = "Accounts may not be added to an administrative workgroup.";
+                return this.RedirectToAction(nameof(Details), new { id = id });
+            }
+
+
+            //chartsString is a stringified json object. deserialized it into a list of WorkgroupFinjectorChart
+            var charts = Newtonsoft.Json.JsonConvert.DeserializeObject<List<WorkgroupFinjectorChart>>(chartsString);
```

Comment:
@srkirkland Seem like the correct way to do this? It seems familiar and does work.

Reply:
Been a long time but seems wrong like it should be a bool value and default to false